### PR TITLE
10 board class

### PR DESCRIPTION
**Pull request by manuelkoso**

### CODE REVIEW

File: src/main/java/Cell.java [hunk 1]
```diff
@@ -24,4 +24,9 @@ public void putStone(@NotNull Stone stone) {
         this.occupied = true;
     }
 
+    public void clear() {
+        this.stone = null;
+        this.occupied = false;
```
Comment (peiva-git) [inline src/main/java/Cell.java:29]: Occupied field may be unnecessary, just perform a check on this.stone != null

File: src/main/java/Board.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    public Iterator<Map.Entry<Position, Cell>> iterator() {
+        return this.cells.entrySet().iterator();
+    }
+
+    public Cell getCell(Position position) throws InvalidBoardPositionException {
+        Cell cell = cells.get(position);
+        if (cell == null) throw new InvalidBoardPositionException("Invalid board position");
+        return cell;
+    }
+
```
Comment (peiva-git) [inline src/main/java/Board.java:38]: A getStone(Position) method may be useful in the Board class

**Review by peiva-git — APPROVED** [review]
Well done!

File: src/main/java/exceptions/InvalidBoardPositionException.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package exceptions;
+
+public class InvalidBoardPositionException extends RuntimeException {
```
Comment (peiva-git) [inline src/main/java/exceptions/InvalidBoardPositionException.java:3]: Is it necessary to have to separate exception classes for an invalid position?

Comment (manuelkoso) [reply src/main/java/exceptions/InvalidBoardPositionException.java:3]: No, you are right but I think that we have to refactor the class name Position to BoardPosition

File: src/main/java/Board.java (hunk 2 above)
Comment (manuelkoso) [reply src/main/java/Board.java:38]: Yes of course, the class implementation is not finished yet

File: src/main/java/Cell.java (hunk 1 above)
Comment (manuelkoso) [reply src/main/java/Cell.java:29]: Yes, I will remove it

**Pull request merged**